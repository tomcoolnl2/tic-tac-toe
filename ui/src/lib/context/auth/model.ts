import { User } from '@tic-tac-toe/model';
import { AuthError } from './error';

export interface AuthContextState {
	fetchUserName: () => Promise<string | AuthError | null>;
	login: (pwd: string) => Promise<AuthError | User>;
	logout: () => Promise<User>;
}
