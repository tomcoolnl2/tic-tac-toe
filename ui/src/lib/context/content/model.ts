import { Locale } from '@tic-tac-toe/model';

export type ContentAction =
	| { type: 'FETCH_START' }
	| { type: 'FETCH_SUCCESS'; payload: AppContent }
	| { type: 'FETCH_ERROR'; payload: Error }
	| { type: 'SET_LANGUAGE'; payload: Locale };

export interface ContentState {
	appContent: AppContent | null;
	isContentLoading: boolean;
	contentError: Error | null;
	locale: Locale;
}

export type ContentStateWithLanguageSelector = ContentState & {
	setLanguage: (locale: Locale) => void;
};

export interface AppContent {
	appTitle: string;
	loginScreen: AppScreenContent;
	settingsScreen: AppScreenContent;
	gameScreen: AppGameContent;
	restartModal: AppScreenContent;
	gameOverModal: AppScreenContent;
}

export interface AppScreenContent {
	title: string | string[];
	subtitle?: string | string[];
	cta1?: string;
	cta2?: string;
	errors?: {
		[key: string]: string;
	};
}

export interface AppGameContent {
	intelligenceLevel: [string, string, string];
	turnIndicator: [string, string];
	scoreBoard: [string, string, string];
}
