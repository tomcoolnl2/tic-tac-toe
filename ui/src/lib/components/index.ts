export { AppLogo } from './logo/logo';
export { Avatar } from './avatar/avatar';
export { Button } from './button/button';
export { Cell } from './cell/cell';
export { ChooseDifficulty } from './choose-difficulty/choose-difficulty';
export { GameOver } from './game-over/game-over';
export { Input } from './input/input';
export { Modal } from './modal/modal';
export { ReloadButton } from './reload-button/reload-button';
export { ScoreBoardItem } from './score-board-item/score-board-item';
export { Spinner } from './spinner/spinner';
export { SymbolChoice } from './symbol-choice/symbol-choice';
export { TurnIndicator } from './turn-indicator/turn-indicator';
