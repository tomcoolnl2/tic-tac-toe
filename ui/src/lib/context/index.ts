export * from './content/content';
export * from './auth/auth';
