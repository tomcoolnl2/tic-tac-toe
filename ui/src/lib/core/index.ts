export * from './canvas';
export * from './grid';
export * from './flex-box';
export * from './divider';
