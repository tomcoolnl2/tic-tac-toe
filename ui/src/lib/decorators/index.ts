export { withTheme } from './withTheme';
export type { ContainerProps } from './withContainer';
export { withContainer } from './withContainer';
