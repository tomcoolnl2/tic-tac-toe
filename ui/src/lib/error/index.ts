export * from './error-boundary';
export { AuthError } from '../context/auth/error';
