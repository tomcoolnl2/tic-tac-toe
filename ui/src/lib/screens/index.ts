export { LoadingScreen } from './loading';
export { LoginScreen } from './login';
export { ErrorScreen } from './error';
export { GameScreen } from './game';
export { SettingsScreen } from './settings';
export { RestartModalScreen } from './reload';
export { GameOverModalScreen } from './game-over';
