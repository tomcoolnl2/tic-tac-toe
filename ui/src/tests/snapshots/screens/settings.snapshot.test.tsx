import { render } from '@testing-library/react';
import { IntelligenceLevel, PlayerSymbol } from '@tic-tac-toe/model';
import { SettingsScreen } from '../../../lib/screens';

describe('SettingsScreen screen component snapshot test', () => {
	const content = {
		cta1: 'Start game',
		title: "Pic Player 1's mark",
		subtitle: 'Remember, X goes first',
	};
	it('should match the snapshot', () => {
		const { asFragment } = render(
			<SettingsScreen
				content={content}
				playerSymbol={PlayerSymbol.X}
				selectedDifficultySetting={IntelligenceLevel.EASY}
				handleLogout={jest.fn()}
			/>
		);
		expect(asFragment()).toMatchSnapshot();
	});
});
