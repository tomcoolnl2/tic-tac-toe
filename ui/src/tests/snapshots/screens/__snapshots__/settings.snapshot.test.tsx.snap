// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsScreen screen component snapshot test should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex-box direction-column justify-content-flex-start align-items-center"
  >
    <div
      class="flex-box direction-row justify-content-flex-start align-items-flex-start spacing-s logo"
    >
      <span
        class="avatar avatar-x avatar-size-m"
      />
      <span
        class="avatar avatar-o avatar-size-m"
      />
    </div>
    <hr
      class="divider full-width horizontal margin-vertical invisible"
    />
    <hr
      class="divider full-width horizontal margin-vertical-l invisible"
    />
    <span>
      Pic Player 1's Mark
    </span>
    <label
      class="choose-player"
      for="choose-player"
    >
      <input
        id="choose-player"
        type="checkbox"
      />
      <span
        class="choose-player-toggle"
      />
      <div
        class="container cols-2 choose-player-symbols"
        style="--row-height: 1fr; --col-width: 1fr;"
      >
        <div
          style="place-self: center;"
        >
          <span
            class="avatar avatar-x avatar-size-m avatar-variant-dark"
          />
        </div>
        <div
          style="place-self: center;"
        >
          <span
            class="avatar avatar-o avatar-size-m avatar-variant-light"
          />
        </div>
      </div>
    </label>
    <sub>
      Remember, X goes first
    </sub>
    <hr
      class="divider full-width horizontal margin-vertical-l invisible"
    />
    <div
      class="flex-box direction-column justify-content-flex-start align-items-center choose-difficulty"
    >
      <div
        class="flex-box direction-row justify-content-flex-start align-items-flex-start"
      >
        <input
          id="master"
          name="level"
          type="radio"
          value="master"
        />
        <label
          for="master"
          title="master"
        />
        <input
          id="novice"
          name="level"
          type="radio"
          value="novice"
        />
        <label
          for="novice"
          title="novice"
        />
        <input
          checked=""
          id="bieber"
          name="level"
          type="radio"
          value="bieber"
        />
        <label
          for="bieber"
          title="bieber"
        />
      </div>
      <hr
        class="divider full-width horizontal margin-vertical-s invisible"
      />
      <sup>
        bieber
      </sup>
    </div>
    <hr
      class="divider full-width horizontal margin-vertical-l invisible"
    />
    <button
      class="button button-secondary"
    >
      Start game 
      <i
        class="icon-play-sign"
      />
    </button>
  </div>
</DocumentFragment>
`;
